examples/impute/plot_multiple_imputation.py

```diff
@@ -0,0 +1,392 @@
+become too small.
+
+An alternative is using the `sklearn.impute.IterativeImputer` to perform
+multiple imputation: a method where every missing value is imputed multiple
+times. The procedure results in multiple datasets where the observed data is
+identical in every dataset, but the imputed data is different.
+All desired steps after imputation are performed on every dataset, such as
+standardization, feature engineering and model fitting.
+
+One final model is obtained by combining the estimates of each model with
+Rubin's pooling rules. These rules assume that the parameters of interest are
+normally distributed which is the case with, for example, estimates of the mean
+and regression coefficients. Other parameters, such as correlation
+coefficients need transformation to suit the assumption of normality.
+If it is not possible to approximate a normal distribution, it is better to use
+robust summary measures such as medians or ranges instead of using Rubin's
+pooling rules. This applies to an estimate like explained variance.
+
+In sum, Rubin's pooling rules are as follows. The overall point estimate after
+multiple imputation (denoted by Qbar) is the average of all the m point
+estimates. The variance of the overall point estimate is a combination of
+so-called within imputation variance (Ubar) and between imputation
+variance (B). Ubar is the average of the m variances of the m point estimates.
+Both Qbar and Ubar are corrected with a factor 1 / m to account for sampling
+variance. The between imputation variance (B) is the sum of the squared
+difference between Qbar and the m point estimates, corrected with a factor
+1 / (m – 1). Then, the total variance (T) of the MI overall point estimate is
+Ubar + B + B/m.
+
+In this Example we will show how to use the `sklearn.impute.IterativeImputer`
+to perform multiple imputation. In Example 1 we show the effect of Rubin’s
+pooling rules on the variance of regression estimates. Due to the between
+imputation variance, the standard errors of all regression coefficients are
+larger with multiple imputation than with single imputation.
+
+In Example 2 we show how to set up a prediction model using multiple
+imputation. We combine the predictions from multiple imputations and show that
+this results in better regression performance.
+
+# Authors: Rianne Schouten <r.m.schouten@uu.nl>
+           Sergey Feldman <sergeyfeldman@gmail.com>
+# License: BSD 3 clause
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import stats, special
+
+from sklearn.datasets import load_boston
+from sklearn.linear_model import RidgeCV
+from sklearn.preprocessing import StandardScaler
+from sklearn.model_selection import train_test_split
+from sklearn.impute import IterativeImputer
+from sklearn.metrics import mean_squared_error as mse
+
+rng = np.random.RandomState(0)
+
+N_ITER = 20  # number of iterations in IterativeImputer
+N_SIM = 3  # number of simulations in Example 2
```

Comment:
For an example, I think that we could be a bit more flexible with PEP8. Move this variable closer to its use even if defined globally.

Reply:
I put it up here so it's easy to tell what you can change to make it run much faster. I'd prefer to leave as is.